File: frontend/src/types/photo.ts
export type Photo = {
  id: number;
  url: string;
  tags: string[];
  title?: string;
  width: number;
  height: number;
};

/**
 * 編集用
 */
export type PhotoDetail = {
  id: number;
  imageUrl: string;
  aspectRatio: number;
  title?: string | null;
  description?: string | null;
  categoryId?: number | null;
  userId?: number | null;
  isVisible: boolean;
  takenAt?: string | null; // ISO文字列（例: "2025-03-30T12:34:56Z"）
  tags?: string[];
};

/**
 * 送信用
 */
export type PhotoUpdateRequest = {
  photo_id: number;
  title: string | null;
  description: string | null;
  category_id: number | null;
  is_visible: boolean;
  taken_at: string | null; // ISO8601
  tags: string[];
};

/** 配列用 */
export type PhotoBulkUpdateRequest = {
  updates: PhotoUpdateRequest[];
};

File: backend/internal/model/photo.go
package model

import "time"

type Photo struct {
	ID          int64  `gorm:"primaryKey"`
	ImageURL    string `gorm:"unique;not null"`
	Width       int
	Height      int
	AspectRatio float64
	Title       *string
	Description *string
	CategoryID  *int64
	UserID      *int64
	IsVisible   bool `gorm:"default:false"`
	TakenAt     *time.Time
	CreatedAt   *time.Time `gorm:"autoCreateTime"`
	UpdatedAt   *time.Time `gorm:"autoUpdateTime"`
	User        *User      `gorm:"foreignKey:UserID"`
	GPS         *PhotoGPS  `gorm:"foreignKey:PhotoID"`
	Exif        *PhotoExif `gorm:"foreignKey:PhotoID"`
	Tags        []*Tag     `gorm:"many2many:photo_tags;"`
}
